src/Service/Media/MediaFileManager.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Service\Media;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class MediaFileManager
+{
+    public function __construct(
+        private string $mediumFolder,
+        private LoggerInterface $logger,
+    ){}
+
+    public function saveFile(?UploadedFile $file,string $newName):void{
+        $file?->move($this->mediumFolder, $newName);
+    }
+
+    public function deleteFile(?string $fileName):void{
+        $filePath = $this->mediumFolder . '/' . $fileName??'';
+        if (!file_exists($filePath)) {
+            return;
+        }
+        try {
+                unlink($filePath) ?? throw new \Exception($filePath);
```

Comment:
```suggestion
                if(false === unlink($filePath)){
                    throw new \Exception($filePath);
                }
```
si la suppression échoue, on renvoi **false** et non null